components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java

```diff
@@ -38,6 +38,7 @@
     public static final String NOTIFICATION_TYPE_ACCOUNT_CONFIRM = "accountconfirmation";
     public static final String NOTIFICATION_TYPE_RESEND_ACCOUNT_CONFIRM = "resendaccountconfirmation";
     public static final String NOTIFICATION_TYPE_EMAIL_CONFIRM = "emailconfirm";
+    public static final String NOTIFICATION_TYPE_TENANT_REGISTRATION_EMAIL_CONFIRM = "tenantRegistrationEmailConfirm";
```

Comment:
This is added mainly to identify the notification template right? Do we need to say that this is EMAIL?

If we enable SMS also as a channel for confirmation, we cannot use this to identify the template because the constant has `email` in its value. As in other places, I think it's better to make it general.

To convert the notification channel we can add the channel to the event properties. WDYT?

This is how the way in self-registration[1] and account recovery.
[1] - https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserSelfRegistrationHandler.java#L409
[2] - https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/confirmation/ResendConfirmationManager.java#L227

Reply:
I don't this we will have a mobile verification for the tenant, but think making it generic will be better? WDYT?